src/main/java/com/linkedin/kafka/clients/largemessage/ConsumerRecordsProcessor.java

```diff
@@ -311,10 +314,12 @@ public void close() {
     K key = _keyDeserializer.deserialize(tp.topic(), consumerRecord.key());
     byte[] valueBytes = parseAndMaybeTrackRecord(tp, consumerRecord.offset(), consumerRecord.value());
     V value = _valueDeserializer.deserialize(tp.topic(), valueBytes);
-    if (value != null) {
+    if (valueBytes != INCOMPLETE_RESULT) {
```

Comment:
V value probably needs to get initialized inside of the check for if (valueBytes != INCOMPLETE_RESULT) since we don't know what the deserializer will do with it; it may throw an exception.